tests/core/pyspec/eth2spec/test/altair/transition/test_slashing.py

```diff
@@ -0,0 +1,69 @@
+import random
+from eth2spec.test.context import (
+    MINIMAL,
+    fork_transition_test,
+    with_presets,
+)
+from eth2spec.test.helpers.constants import PHASE0, ALTAIR
+from eth2spec.test.helpers.fork_transition import (
+    do_altair_fork,
+    state_transition_across_slots_with_ignoring_proposers,
+    transition_until_fork,
+)
+from eth2spec.test.helpers.random import (
+    slash_random_validators,
+)
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=1)
+@with_presets([MINIMAL], reason="only test with non-full committee")
+def test_transition_with_one_fourth_slashed_active_validators_pre_fork(state,
+                                                                       fork_epoch,
+                                                                       spec,
+                                                                       post_spec,
+                                                                       pre_tag,
+                                                                       post_tag):
+    """
+    1/4 validators are slashed but still active at the fork transition.
+    """
+    # slash 1/4 validators
+    slashed_indices = slash_random_validators(spec, state, rng=random.Random(5566), fraction=0.25)
+    assert len(slashed_indices) > 0
+
+    # check if some validators are slashed but still active
+    for validator_index in slashed_indices:
+        validator = state.validators[validator_index]
+        assert validator.slashed
+        assert spec.is_active_validator(validator, spec.get_current_epoch(state))
+    assert not spec.is_in_inactivity_leak(state)
+
+    transition_until_fork(spec, state, fork_epoch)
+
+    assert spec.get_current_epoch(state) < fork_epoch
+
+    yield "pre", state
+
+    # irregular state transition to handle fork:
+    state, _ = do_altair_fork(state, spec, post_spec, fork_epoch, with_block=False)
```

Comment:
because `slash_random_validators` may select the next block proposer. I didn't want to change `slash_random_validators` too much. Do you feel strongly that we should add an `avoid_next_proposer` flag to `slash_random_validators` helper?